src/HotChocolate/Core/src/Execution/Processing/MiddlewareContext.Pure.cs

```diff
@@ -75,12 +76,16 @@ public void Clear()
 
         public T Parent<T>()
         {
-            if (_parent is T casted)
+            return _parent switch
             {
-                return casted;
-            }
-
-            return default!;
+                T casted => casted,
+                null => default!,
+                _ => throw ResolverContext_CannotCastParent(
+                    Selection.Field.Coordinate,
+                    _path,
+                    typeof(T),
+                    _parent.GetType())
+            };
```

Comment:
Should we do cast + try catch here? Would this be more optimized than `is T` ?

Reply:
I think we tried it ... and it was I think, the only issue was in a non-pure resolver we have conversion logic on top.

Reply:
There are a couple of questions around this ... should we adopt the logic of the async resolver with type conversion or keep it simple like it is. But in any case ... Benchmarks first. This means we do it in a separate PR.